spec/models/pubmed_source_record_spec.rb

```diff
@@ -264,37 +50,33 @@ def check_author_hash(author)
     end
   end
 
-  describe '.source_as_hash' do
-    context 'DOI extraction' do
-      def doi(pmid)
-        record = described_class.send(:get_pubmed_record_from_pubmed, pmid)
-        return nil if record.nil?
-
-        record.source_as_hash[:identifier].find { |id| id[:type] == 'doi' }
-      end
-      it 'constructs a URL based on the DOI' do
-        expect(doi(12_529_422)).to include(url: 'https://doi.org/10.1091/mbc.e02-06-0327')
-      end
-
-      context 'extracts from ArticleId' do
-        it 'works when ELocationID is missing' do
-          expect(doi(12_529_422)).to include(id: '10.1091/mbc.e02-06-0327')
-        end
-
-        it 'works when ELocationID is present' do
-          expect(doi(23_453_302)).to include(id: '10.1016/j.neunet.2013.01.016')
-        end
-
-        it 'works when record is longer than 64kb' do
-          expect(doi(26_430_984)).to include(id: '10.1103/PhysRevLett.115.121604')
-        end
-      end
-
-      context 'extracts from ELocationID' do
-        it 'works when ArticleId is missing' do
-          expect(doi(26_858_277)).to include(id: '10.1136/bmj.i493')
-        end
-      end
+  describe '#source_as_hash' do
+    it 'returns a pub_hash' do
+      record = described_class.send(:get_pubmed_record_from_pubmed, pmid_created_1999)
+      expect(record.source_as_hash).to eq(
+        { provenance: 'pubmed', pmid: '10000166',
+          title: 'Fluid permeability in porous media: Comparison of electrical estimates with hydrodynamical calculations.',
```

Comment:
this verifies that we can still call `source_as_hash` on the source record and get the pub hash back, as this method is used throughout the code